src/Features/CSharp/Portable/UseLocalFunction/CSharpUseLocalFunctionDiagnosticAnalyzer.cs

```diff
@@ -358,6 +360,39 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext, INamedTyp
             return false;
         }
 
+        private static bool CanReplaceDelegateWithLocalFunction(
+            INamedTypeSymbol delegateType, 
+            LocalDeclarationStatementSyntax localDeclaration,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            var delegateContainingType = delegateType.ContainingType;
+            if (delegateContainingType is null || !delegateContainingType.IsGenericType)
+            {
+                return true;
+            }
+
+            var delegateTypeParamNames = delegateType.GetAllTypeParameters().Select(p => p.Name).ToImmutableHashSet();
+            var localEnclosingSymbol = semanticModel.GetEnclosingSymbol(localDeclaration.SpanStart, cancellationToken);
+            while (localEnclosingSymbol != null)
+            {
+                if (localEnclosingSymbol.Equals(delegateContainingType))
+                {
+                    return true;
+                }
+
+                var typeParams = localEnclosingSymbol.GetTypeParameters();
+                if (typeParams.Any())
+                {
+                    return !typeParams.Any(p => delegateTypeParamNames.Contains(p.Name));
```

Comment:
>return [](start = 20, length = 6)

Feels like this exits too early if there is deep nesting with each layer adding new type parameters.
I would have thought we'd check this layer for problems and if no problem then we'd continue checking containing types.

Reply:
➡️ Fixed in 44e67c5796d99ff0dccda995063e0f1092b9cd6f with a new test `TestNotAvailableIfTypeParameterChanged4` added